Detectors/ITSMFT/ITS/RawPixelReaderWorkflow/src/RawPixelReaderSpec.cxx

```diff
@@ -0,0 +1,608 @@
+
+  if (TimeCounter == 0) {
+    for (int i = 0; i < NowFolderNames.size(); i++) {
+      if (DiffFileNames[i].size() > 0)
+        TimeCounter = 1;
+    }
+  }
+  if (TimeCounter == 1) {
+    start = std::chrono::high_resolution_clock::now();
+    TimeCounter = 2;
+    cout << "STARTED CLOCK" << endl;
+  }
+
+  for (int i = FolderNames.size(); i < NowFolderNames.size(); i++) {
+    DiffFileNames.push_back(NowFileNames[i]);
+    cout << "New File Size Between New and Initial Runs " << DiffFileNames[i].size() << endl;
+  }
+
+  LOG(DEBUG) << "Total New Files = " << DiffFileNames.size();
+
+  LOG(DEBUG) << "DONE Creating Difference";
+
+  LOG(DEBUG) << "DiffFileNames Size = " << DiffFileNames.size();
+
+  LOG(DEBUG) << "DONE Checking -- Reseting Vectors to the latest vector";
+
+  LOG(DEBUG) << "DONE Updateing Vectors";
+
+  LOG(DEBUG) << "Start Loop";
+
+  for (int i = 0; i < NowFolderNames.size(); i++) {
+
+    LOG(DEBUG) << "i = " << i << "    DiffFileNames[i].size() = " << DiffFileNames[i].size();
+    pos = NowFolderNames[i].find_last_of("/");
+
+    if (pos != string::npos)
+      RunID = NowFolderNames[i].substr(pos + 1);
+    cout << "FileDone = " << FileDone << endl;
+
+    //	for(int j = 0; j < DiffFileNames[i].size(); j++){
+    if (DiffFileNames[i].size() > 0 && FileDone == 1) {
+
+      FileRemain = DiffFileNames[i].size();
+      FileDone = 0;
+      cout << "RunID = " << RunID << endl;
+      cout << "File Location = " << DiffFileNames[i][0] << endl;
+
+      size_t last_index1 = RunID.find_last_not_of("0123456789");
+      string RunIDS = RunID.substr(last_index1 + 1);
+
+      string FileIDS;
+      pos = DiffFileNames[i][0].find_last_of("/");
+      if (pos != string::npos)
+        FileIDS = DiffFileNames[i][0].substr(pos + 1);
+
+      cout << "Before FileIDS = " << FileIDS << endl;
+
+      size_t last_index2 = FileIDS.find_last_not_of("0123456789");
+      FileIDS = FileIDS.substr(last_index2 + 1);
+
+      RunName = std::stoi(RunIDS);
+      FileID = std::stoi(FileIDS);
+
+      ofstream fout(Form("ErrorData/ErrorLogRun%d_File%d.dat", RunName, FileID));
+      fout << " START OF ERROR REPORT For Run " << RunName << "  File " << FileID << endl;
+
+      inpName = DiffFileNames[i][0];
+
+      //		pc.outputs().snapshot(Output{ "TST", "Run", 0, Lifetime::Timeframe }, RunID.data());
+      //		pc.outputs().snapshot(Output{ "TST", "File", 0, Lifetime::Timeframe }, inpName.data());
+
+      EventRegistered = 0;
+      LOG(DEBUG) << "inpName = " << inpName;
+
+      //Inject fake thing Now//
+
+      if (NewFileInj == 1) {
+        cout << "New File Injected, Now Updating the Canvas and Light" << endl;
+        mDigitsTest.emplace_back(0, 0, 0, 0, 0);
+        mMultiDigitsTest.push_back(mDigitsTest[0]);
+        ErrorVecTest.push_back(Error);
+        FileDone = 1;
+        FileInfo = FileDone + FileRemain * 10;
+        pc.outputs().snapshot(Output{ "TST", "Run", 0, Lifetime::Timeframe }, RunName);
```

Comment:
Why do you use TST as a producer? What is the meaning of mDigitsTest and mMultiDigitsTest?

Reply:
TST send arrays of integers to the QC. I think that is the only it works for the moment. I tried what Giulio suggests and it does not work. mDigitsTest is the fake digit that start the QC immediately after we find a new file in the working folder.

Reply:
But why do you need a producer different from ITS, moreover under such a generic name TST/run? What exactly does not work?

Reply:
Hi Ruben,

It looks like it does not work if I use ITS. I mean it looks like ITS can only send digits or clusters. I can have a try over the weekend and see if it works.

Reply:
@MYOMAO Can you explain sending of which object does not work?

Reply:
So I tried to send the int array Error[11] from the RawPixelReaderSpec to the QC but it does not seem to work with ITS

Reply:
@MYOMAO do you mean the message is sent with TST but not with ITS data origin? I don't think it is possible, the origin is just a 4 bytes long identifier in the message header,  it cannot affect message transmission. There should be something else. Could you post here what are the symptoms of message sending not working?

Reply:
@shahor02 OK I can have a try and see if it will work. Thank you!

Reply:
@shahor02 OK! I took you suggestions and it actually worked out! So I have made another commit. Can you take a look at what else I need to do? Thank you very much!